tests: limit code coverage comments to 10

tomasmatus opened the pull request:
fixes: #22071

cockpituous commented on pkg/users/account-create-dialog.js:
```diff
@@ -147,6 +147,9 @@ function AccountCreateBody({ state, errors, change, shells }) {
 function validate_username(username, accounts) {
+    // NOTE: force code coverage comment
+    if (username === "tomuser")
+        return "Nothing happened";
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on pkg/users/account-create-dialog.js:
```diff
@@ -173,6 +176,9 @@ function validate_real_name(real_name) {
 function validate_uid(uid, accounts, min_uid, max_uid, change) {
+    // NOTE: force code coverage comment
+    if (uid === "45551")
+        return "Nothing happened";
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on pkg/users/account-create-dialog.js:
```diff
@@ -197,13 +203,19 @@ function validate_home_dir(dir, directoryExpected) {
 function validate_password(password) {
+    // NOTE: force code coverage comment
+    if (password === "tompassword")
+        return "Nothing happened";
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on pkg/users/account-create-dialog.js:
```diff
@@ -197,13 +203,19 @@ function validate_home_dir(dir, directoryExpected) {
 function validate_password_confirm(password_confirm, password) {
+    // NOTE: force code coverage comment
+    if (password === "tompassword")
+        return "Nothing happened";
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous reviewed (commented):
There are more than 10 code coverage comments, see the full report at None/Coverage/index.html

tomasmatus commented:
ah well this works quite nicely except that the `TEST_ATTACHMENTS_URL` isnt set so the link to full report doesn't work

martinpitt commented on test/common/lcov.py:
```diff
@@ -482,9 +482,13 @@ def limit_comments(comments: list[dict[str, str]]) -> tuple[list[dict[str, str]]
-    report_link_comment = f"There are more than 10 code coverage comments, see the full report at {coverage_report}"
+    log_url = os.environ.get("COCKPIT_CI_LOG_URL", None)
+    if log_url is not None:
+        base_url = log_url.replace("log.html", "")
```
I'd prefer to make this a bit more natural using `os.path.dirname(log_url)`

That also fixes the double slash in the generated URL: https://cockpit-logs.us-east-1.linodeobjects.com/pull-22180-9baec84a-20250707-123857-fedora-42-devel//Coverage/index.html

martinpitt commented on test/common/lcov.py:
```diff
@@ -482,9 +482,13 @@ def limit_comments(comments: list[dict[str, str]]) -> tuple[list[dict[str, str]]
     if len(comments) <= 4:
         return comments, ""
 
-    ta_url = os.environ.get("TEST_ATTACHMENTS_URL", None)
```
Please prefix this commit with "test:" and add a description that TEST_ATTACHMENTS_URL isn't being set any more. This is a bug fix, not something that actually was broken by or is related to your "limit to 10", right?

tomasmatus replied:
this bit is new code. I'll add a bugfix commit for the `flush_cur_comment` function where same env var is also used

martinpitt reviewed (changes requested):
This looks quite nice! One detail to fix, and of course please remove the test commit. The coverage report is currently inaccessible, that's my fault. I'll re-fix that in the Linode config.

cockpituous reviewed (commented):
There are more than 10 code coverage comments, see [the full report here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-22180-9baec84a-20250707-123857-fedora-42-devel//Coverage/index.html).

martinpitt commented:
> The coverage report is currently inaccessible, that's my fault

 Ah no, it's not -- https://cockpit-logs.us-east-1.linodeobjects.com/pull-22180-9baec84a-20250707-123857-fedora-42-devel/Coverage/index.html works fine. So it's the double slash.

tomasmatus commented:
Yes, the double slash is the problem. I also need to look at the comment delete part of this because the main review body didn't get deleted. Only comments on specific line did

tomasmatus commented:
> I also need to look at the comment delete part of this because the main review body didn't get deleted. Only comments on specific line did

Seems like that is not possible according to [docs](https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#delete-a-pending-review-for-a-pull-request)

> Submitted reviews cannot be deleted.

tomasmatus commented:
Great stuff, works. I made one change and the review comment will now link directly to the PR diff coverage report `/Coverage/lcov/github-pr.diff.gcov.html` instead of the main `/Coverage/index.html`.  @martinpitt what do you think?

The individual comments are also fixed and have the `Detail` link.

I changed the comment limit back to 10 (from 4 for testing) and dropped the debug commit. Triggering fedora-coreos/other.

martinpitt reviewed (approved):
Excellent, thanks @tomasmatus !
